lib/maybe.mjs

```diff
@@ -0,0 +1,26 @@
+import curry from './curry.mjs'
+
+const maybe = x => {
+  const contents = x
+
+  const isNothing = () =>
+    contents === null || contents === undefined
+
+  const map = fn =>
+    maybe(isNothing() ? null : fn(contents))
+
+  const open = curry(
+    (isEmpty, fn) =>
+      isNothing()
+        ? isEmpty
```

Comment:
Instead of an `isEmpty` value, should it call a provided function?

Reply:
I'm doing some research on maybes now, and I think a value is correct here, but we should name it something like `default` instead of `isEmpty`.

Reply:
It should be clear to the user that this value will be sent back when `maybe` is not a value. Would `fallbackValue` work better?

Reply:
Yeah. I would even be down for a ridiculously long name here like `defaultValueForEmptyCase`.